user_class/user.rb

```diff
@@ -1,2 +1,105 @@
 class User
+  @@VALID_ATR = %i(first_name last_name email age address)
+
+  attr_reader :all
+
+  def initialize
+    @all = []
+  end
+
+  def create(id:, first_name:, last_name:, email:, **kwargs)
+    info = {
+      id: id,
+      first_name: first_name,
+      last_name: last_name,
+      email: email
+    }
+    if kwargs[:age]
+      info.merge!({age: kwargs[:age]})
+    end
+    if kwargs[:address]
+      info.merge!({address: kwargs[:address]})
+    end
+
+    @all << info
+  end
+
+  def count
+    @all.size
+  end
+
+  def find(id)
+    @all.find { |user| user if id == user[:id] }
+  end
+
+  def where(**kwargs)
+    elements = []
+
+    @all.each do |user|
+      check = true
+      kwargs.each do |atr, value|
+        check = false unless user[atr] == value
+      end
+      elements << user if check
+    end
+
+    elements
+  end
+
+  def update(id:, **info_update)
+    @all.map do |user|
+      if id == user[:id]
+        info_update.each do |atr, value_update|
+          user[atr] = value_update if @@VALID_ATR.include?(atr)
+        end
+      end
+    end
+  end
+
+  def destroy(id:)
+    @all.delete_if { |user| id == user[:id] }
+  end
 end
+
+users = User.new
+user1_info = {
+  id: '0001',
+  first_name: 'Axel',
+  last_name: 'Ryzen',
+  email: 'aryzen@example.com'
+}
+users.create(**user1_info)
+user2_info = {
```

Comment:
Just as an style suggestions. Leave an empty line before this line to make the code more readable.